reflect/src/main/java/dagger/reflect/ReflectiveModuleParser.java

```diff
@@ -42,7 +42,7 @@ static void parse(Class<?> moduleClass, @Nullable Object instance,
           throw new IllegalArgumentException("Private module methods are not allowed: " + method);
         }
 
-        Binding<?> binding;
+        Binding binding;
```

Comment:
```suggestion
        UnlinkedBinding binding;
```
note: needs import

Reply:
Inserting something that's already linked would be valid. There may not be any now, but there could be in the future.